aci/aci.go

```diff
@@ -64,12 +66,47 @@ func createACIContainers(ctx context.Context, aciContext store.AciContext, group
 	return createOrUpdateACIContainers(ctx, aciContext, groupDefinition)
 }
 
+func autocreateFileshares(ctx context.Context, project *types.Project) error {
+	if len(project.Volumes) == 0 {
+		return nil
+	}
+	for _, v := range project.Volumes {
+		if v.Driver != convert.AzureFileDriverName {
+			return fmt.Errorf("cannot use ACI volume, required driver is %q, found %q", convert.AzureFileDriverName, v.Driver)
+		}
+		shareName, ok := v.DriverOpts[convert.VolumeDriveroptsShareNameKey]
+		if !ok {
+			return fmt.Errorf("cannot retrieve fileshare name for Azurefile")
+		}
+		accountName, ok := v.DriverOpts[convert.VolumeDriveroptsAccountNameKey]
+		if !ok {
+			return fmt.Errorf("cannot retrieve account name for Azurefile")
+		}
+		clt, err := client.New(ctx)
+		if err != nil {
+			return err
+		}
```

Comment:
You could do this once outside the loop

Reply:
I don't think so, you can have multiple volumes, you need to check for each if you need to create it or not

Reply:
I meant creating the client. Since it's created from the context without any volume information, we can move it outside the loop.

Reply:
done